public/app/features/inspector/InspectDataTab.test.tsx

```diff
@@ -1,9 +1,13 @@
 import React, { ComponentProps } from 'react';
-import { FieldType, DataFrame } from '@grafana/data';
+import { DataFrame, FieldType } from '@grafana/data';
 import { render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import { InspectDataTab } from './InspectDataTab';
 
+// the mock below gets rid of this warning from recompose:
+// Warning: React.createFactory() is deprecated and will be removed in a future major release. Consider using JSX or use React.createElement() directly instead.
+jest.mock('@jaegertracing/jaeger-ui-components', () => ({}));
```

Comment:
InspectDataTab is not importing jeager components, so is this not strange? Will all tests need this? If so maybe we need to add this to jest setup?

Reply:
Hmm, good call. I didn't muster to trace where we import `@jaegertracing/jaeger-ui-components` because of all the circular dependencies. So far only 2 tests show this, if I see more I'll add it to jest setup.

Reply:
@torkelo Adding this to setup would make these tests that depend on @jaegertracing/jaeger-ui-components fail:
- public/app/features/explore/TraceView/TraceView.test.tsx
- public/app/features/explore/TraceView/useDetailState.test.ts
- public/app/features/explore/TraceView/useSearch.test.ts

So I'm reverting back to adding `jest.mock('@jaegertracing/jaeger-ui-components', () => ({}));` to the 2 tests that need it. I think it's better that @grafana/observability-logs-and-traces look into a long term solution for this warning from `recompose`.

Reply:
:+1:

Reply:
I'm working on updating the `SpanBar` component that is causing this error as part of the same chore